zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -735,12 +764,30 @@ public void incInProcess() {
 
     public void decInProcess() {
         requestsInProcess.decrementAndGet();
+        if (requestThrottler != null) {
+            requestThrottler.throttleWake();
+        }
     }
 
     public int getInProcess() {
         return requestsInProcess.get();
     }
 
+    public int getInflight() {
+        if (RequestThrottler.enabled()) {
+            return requestThrottleInflight();
+        } else {
+            return getInProcess();
+        }
+    }
+
+    public int requestThrottleInflight() {
```

Comment:
consider make this private as it's not called elsewhere.

Reply:
done